src/classes/ALLO_Allocations_TDTM.cls

```diff
@@ -90,26 +83,27 @@ public class ALLO_Allocations_TDTM extends TDTM_Runnable {
         }
 
         // dispatch to the correct handler
-        if (objResult.getsObjectType() == Allocation__c.sObjectType && !hasRunForAllocationsBefore && 
+        if (objResult.getsObjectType() == Allocation__c.sObjectType && !ALLO_ProcessControl.checkRecursionState() &&
             (triggerAction == TDTM_Runnable.Action.BeforeInsert || triggerAction == TDTM_Runnable.Action.BeforeUpdate)) {
 
             runAllocationBeforeTriggerHandler(newlist, oldlist, triggerAction, objResult);
-            hasRunForAllocationsBefore = true;
-        } else if (objResult.getsObjectType() == Allocation__c.sObjectType && !hasRunForAllocationsAfter && 
+        } else if (objResult.getsObjectType() == Allocation__c.sObjectType && !ALLO_ProcessControl.checkRecursionState() &&
```

Comment:
I know there was a reason I pulled that out, but I want to go back and re-test the logic. The performance results for the ALLO tests show an increase of one DML statement for a few of the ALLO unit tests, so I want to verify that I didn't miss something in the initial review.